app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java

```diff
@@ -459,12 +459,13 @@ private PaymentMethod mapPaymentMethods(PaymentMethodEntity paymentMethod,
           .equals(availablePaymentMethod.getId())) {
         PaymentMethodFee paymentMethodFee = mapPaymentMethodFee(availablePaymentMethod.getFee());
         return new PaymentMethod(paymentMethod.getId(), paymentMethod.getLabel(),
-            paymentMethod.getIconUrl(), paymentMethod.getAsync(), paymentMethodFee, true, null);
+            paymentMethod.getIconUrl(), paymentMethod.getAsync(), paymentMethodFee, true, null,
+            false);
       }
     }
     PaymentMethodFee paymentMethodFee = mapPaymentMethodFee(paymentMethod.getFee());
     return new PaymentMethod(paymentMethod.getId(), paymentMethod.getLabel(),
-        paymentMethod.getIconUrl(), paymentMethod.getAsync(), paymentMethodFee, false, null);
+        paymentMethod.getIconUrl(), paymentMethod.getAsync(), paymentMethodFee, false, null, true);
```

Comment:
Can you explain how does this work as expected? From what I'm reading from the code, you're simply saying that if a payment method is not available, then this unavailable payment method will show a top up button. But shouldn't the rule be: only show the topup button if the only available payment method is APPC-C?